### ISSUE TITLE
PR creation fails after moving components

### ISSUE THREAD

**Issue by dsiebel** [google-github-actions/release-please-action#704]
### TL;DR

We are using manifest releaser on a repository containing github actions and reusable workflows that are supposed to be used by referencing the files in the repo directly.

After moving a component from `.` to a sub-directory, creating the PR for that component results in 
> release-please failed: Error while creating a tree and updating the ref: Resource not accessible by integration

### Expected behavior

Release PR gets created		

### Observed behavior

Release PR creation fails

### Action YAML

```yaml
on:
  push:
    branches:
      - 'main'
name: 'release-please'
jobs:
  release-please:
    name: 'Release Please'
    runs-on: 'ubuntu-latest'
    steps:
      - name: 'manifest(main)'
        uses: 'google-github-actions/release-please-action@5c07f8be172b1f6e90f9c35baf8184461b91b85f' #v3.7.1
        with:
          command: manifest
          default-branch: 'main'
```

### Log output

```text
Fetching release-please-config.json from branch main
❯ Fetching .release-please-manifest.json from branch main
✔ Building pull requests
✔ Building strategies by path
❯ .github/workflows: simple
❯ .github/actions/setup: simple
❯ .github/actions/auth: simple
❯ .github/actions/containerize: simple
❯ .github/actions/render: simple
❯ .github/actions/sync: simple
✔ Collecting release commit SHAs
❯ release search depth: 400
❯ Fetching releases with cursor undefined
❯ Found release for path .github/actions/sync, action-sync-v1.0.2
❯ Found release for path .github/actions/render, action-render-v1.0.1
❯ Found release for path .github/actions/containerize, action-containerize-v1.0.1
❯ Found release for path .github/actions/auth, action-auth-v1.0.1
❯ Found release for path .github/workflows, workflows-v1.0.5
❯ Found release for path .github/actions/setup, action-setup-v1.0.0
❯ release for path: .github/actions/sync, version: 1.0.2, sha: be244d18816635b0065a69dcdb2beae153bb42f2
❯ release for path: .github/actions/render, version: 1.0.1, sha: be244d18816635b0065a69dcdb2beae153bb42f2
❯ release for path: .github/actions/containerize, version: 1.0.1, sha: be244d18816635b0065a69dcdb2beae153bb42f2
❯ release for path: .github/actions/auth, version: 1.0.1, sha: be244d18816635b0065a69dcdb2beae153bb42f2
❯ release for path: .github/workflows, version: 1.0.5, sha: be244d18816635b0065a69dcdb2beae153bb42f2
❯ release for path: .github/actions/setup, version: 1.0.0, sha: 899ccdec65caf2a1eb2879e4ff856e942f0f5e34
✔ Collecting commits since all latest releases
❯ commit search depth: 500
❯ Set(2) {
  'be244d18816635b0065a69dcdb2beae153bb42f2',
  '899ccdec65caf2a1eb2879e4ff856e942f0f5e34'
}
❯ Fetching merge commits on branch main with cursor: undefined
❯ Backfilling file list for commit: a0d3b253a71cc94d247795b88030714b4136ad3a
❯ Found 1 files
❯ Backfilling file list for commit: 842d01a980fe36dcf2bc62c5b4a54bf49593f03c
❯ Found 3 files
❯ Backfilling file list for commit: b80379601852dbf5b453a04c8630089db33f5d7f
❯ Found 1 files
❯ Backfilling file list for commit: 48f1212f70c58fd7d29e0fd8b9c4591701ac580d
❯ Found 1 files
❯ Backfilling file list for commit: a48caa079aade69de12c443044e8c83aa61a327e
❯ Found 1 files
❯ Backfilling file list for commit: d64e39f12a3d9fb5e233ea32519c398e6c6579ec
❯ Found 1 files
❯ Backfilling file list for commit: cf4190c66f10de2faffd8752aa69f9de66c43350
❯ Found 1 files
❯ Backfilling file list for commit: 4a32ece4cfd722d53c2d237cf5bcdcbd63894d89
❯ Found 1 files
❯ Backfilling file list for commit: 5842d808734191e7823f2fea3d2bea95ff283428
❯ Found 1 files
❯ Backfilling file list for commit: 413000c06c573e2c7848938e4b8ea86a4f4f2164
❯ Found 1 files
❯ Backfilling file list for commit: e66eab5b55b7ceaafd7c1e457820a1ed9b931e87
❯ Found 1 files
❯ Backfilling file list for commit: cec4b8535c61297b95397413b3cfcc97e3b0a1be
❯ Found 1 files
❯ Backfilling file list for commit: ac5b921c4e450c06035a1c4a0a7eed0e553ae878
❯ Found 1 files
❯ Backfilling file list for commit: 101ac24a8731cae9dcd6f23e913cfb62562da07e
❯ Found 1 files
❯ Backfilling file list for commit: b22b381efbc3cb9f55ed624f4656252b2f00e756
❯ Found 4 files
❯ Backfilling file list for commit: ea849ef25a680095214986cb4efdd832c6a477ee
❯ Found 1 files
✔ Splitting 19 commits by path
✔ Building candidate release pull request for path: .github/workflows
❯ type: simple
❯ targetBranch: main
❯ commits: 6
✔ Considering: 6 commits
❯ component: workflows
❯ pull request title pattern: undefined
✔ Building candidate release pull request for path: .github/actions/setup
❯ type: simple
❯ targetBranch: main
❯ commits: 0
✔ Considering: 0 commits
✔ No commits for path: .github/actions/setup, skipping
✔ Building candidate release pull request for path: .github/actions/auth
❯ type: simple
❯ targetBranch: main
❯ commits: 0
✔ Considering: 0 commits
✔ No commits for path: .github/actions/auth, skipping
✔ Building candidate release pull request for path: .github/actions/containerize
❯ type: simple
❯ targetBranch: main
❯ commits: 0
✔ Considering: 0 commits
✔ No commits for path: .github/actions/containerize, skipping
✔ Building candidate release pull request for path: .github/actions/render
❯ type: simple
❯ targetBranch: main
❯ commits: 0
✔ Considering: 0 commits
✔ No commits for path: .github/actions/render, skipping
✔ Building candidate release pull request for path: .github/actions/sync
❯ type: simple
❯ targetBranch: main
❯ commits: 0
✔ Considering: 0 commits
✔ No commits for path: .github/actions/sync, skipping
✔ Looking for open release pull requests
✔ found 0 open release pull requests.
✔ Looking for snoozed release pull requests
✔ found 0 snoozed release pull requests.
❯ Fetching .github/workflows/CHANGELOG.md from branch main
❯ Fetching .github/workflows/version.txt from branch main
⚠ file .github/workflows/version.txt did not exist
❯ Fetching .release-please-manifest.json from branch main
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "a0d3b253a71cc94d247795b88030714b4136ad3a".
✔ Successfully created branch at https://api.github.com/repos/trivago/github-workflows-skaffold/git/refs/heads/release-please--branches--main--components--workflows
✔ Got the latest commit tree
✖ RequestError [HttpError]: Error while creating a tree and updating the ref: Resource not accessible by integration
    at /home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:16555:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async createTree (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:22120:22)
Error: release-please failed: Error while creating a tree and updating the ref: Resource not accessible by integration
    at async commitAndPush (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:22167:29)
    at async createPullRequest (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:22771:5)
    at async /home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:85454:30
    at async GitHub.createPullRequest (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:86530:20)
    at async Manifest.createOrUpdatePullRequest (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:87059:32)
    at async Promise.all (index 0)
    at async Manifest.createPullRequests (/home/runner/work/_actions/google-github-actions/release-please-action/5c07f8be172b1f6e90f9c35baf8184461b91b85f/dist/index.js:86999:33) {
  status: 403,
  response: {
    url: 'https://api.github.com/repos/trivago/github-workflows-skaffold/git/trees',
    status: 403,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      connection: 'close',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Wed, 25 Jan 2023 10:48:57 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-api-version-selected': '2022-11-28',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '2014:5481:1B60800:1C6A6E6:63D10918',
      'x-ratelimit-limit': '15000',
      'x-ratelimit-remaining': '14971',
      'x-ratelimit-reset': '1674647325',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '29',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Resource not accessible by integration',
      documentation_url: 'https://docs.github.com/rest/reference/git#create-a-tree'
    }
  },
  request: {
    method: 'POST',
    url: 'https://api.github.com/repos/trivago/github-workflows-skaffold/git/trees',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/19.0.4 octokit-core.js/4.0.5 Node.js/16.16.0 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"tree":[{"path":".github/workflows/CHANGELOG.md","mode":"100644","type":"blob","content":"# Changelog\\n\\n## [1.0.6](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.5...workflows-v1.0.6) (2023-01-25)\\n\\n\\n### Bug Fixes\\n\\n* bump sync action to v1.0.1 ([d64e39f](https://github.com/trivago/github-workflows-skaffold/commit/d64e39f12a3d9fb5e233ea32519c398e6c6579ec))\\n* remove trailing slashes on directory defaults ([4a32ece](https://github.com/trivago/github-workflows-skaffold/commit/4a32ece4cfd722d53c2d237cf5bcdcbd63894d89))\\n* use commit SHAs to avoid the versions kerfuffle ([cec4b85](https://github.com/trivago/github-workflows-skaffold/commit/cec4b8535c61297b95397413b3cfcc97e3b0a1be))\\n* use correct action tags (monorepo) ([ac5b921](https://github.com/trivago/github-workflows-skaffold/commit/ac5b921c4e450c06035a1c4a0a7eed0e553ae878))\\n\\n\\n### Miscellaneous\\n\\n* bump sync action to v1.0.1 ([cf4190c](https://github.com/trivago/github-workflows-skaffold/commit/cf4190c66f10de2faffd8752aa69f9de66c43350))\\n\\n## [1.0.5](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.4...workflows-v1.0.5) (2023-01-24)\\n\\n\\n### Bug Fixes\\n\\n* use bash to take care of input sanitization ([#18](https://github.com/trivago/github-workflows-skaffold/issues/18)) ([2a9d930](https://github.com/trivago/github-workflows-skaffold/commit/2a9d930b47337890ceb3fdc362b3ebd64055438a))\\n\\n## [1.0.4](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.3...workflows-v1.0.4) (2023-01-23)\\n\\n\\n### Bug Fixes\\n\\n* bump sync action to v1.0.1 ([d64e39f](https://github.com/trivago/github-workflows-skaffold/commit/d64e39f12a3d9fb5e233ea32519c398e6c6579ec))\\n\\n\\n### Miscellaneous\\n\\n* bump sync action to v1.0.1 ([cf4190c](https://github.com/trivago/github-workflows-skaffold/commit/cf4190c66f10de2faffd8752aa69f9de66c43350))\\n\\n\\n### Documentation\\n\\n* adjust examples for workflows-v1.0.4 release ([a48caa0](https://github.com/trivago/github-workflows-skaffold/commit/a48caa079aade69de12c443044e8c83aa61a327e))\\n\\n## [1.0.3](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.2...workflows-v1.0.3) (2023-01-23)\\n\\n\\n### Bug Fixes\\n\\n* remove trailing slashes on directory defaults ([4a32ece](https://github.com/trivago/github-workflows-skaffold/commit/4a32ece4cfd722d53c2d237cf5bcdcbd63894d89))\\n* use `status --porcelain` to detect changes to the local working tree ([5842d80](https://github.com/trivago/github-workflows-skaffold/commit/5842d808734191e7823f2fea3d2bea95ff283428))\\n\\n\\n### Documentation\\n\\n* **examples:** fix links and tag names ([e66eab5](https://github.com/trivago/github-workflows-skaffold/commit/e66eab5b55b7ceaafd7c1e457820a1ed9b931e87))\\n* **examples:** reference correct workflow file ([413000c](https://github.com/trivago/github-workflows-skaffold/commit/413000c06c573e2c7848938e4b8ea86a4f4f2164))\\n\\n## [1.0.2](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.1...workflows-v1.0.2) (2022-12-21)\\n\\n\\n### Bug Fixes\\n\\n* use commit SHAs to avoid the versions kerfuffle ([cec4b85](https://github.com/trivago/github-workflows-skaffold/commit/cec4b8535c61297b95397413b3cfcc97e3b0a1be))\\n\\n## [1.0.1](https://github.com/trivago/github-workflows-skaffold/compare/workflows-v1.0.0...workflows-v1.0.1) (2022-12-21)\\n\\n\\n### Bug Fixes\\n\\n* use correct action tags (monorepo) ([ac5b921](https://github.com/trivago/github-workflows-skaffold/commit/ac5b921c4e450c06035a1c4a0a7eed0e553ae878))\\n\\n## 1.0.0 (2022-12-21)\\n\\n\\n### Features\\n\\n* gitops-github-release composed workflow (v1-rc1) ([#1](https://github.com/trivago/github-workflows-skaffold/issues/1)) ([8f7eeed](https://github.com/trivago/github-workflows-skaffold/commit/8f7eeed1393bbb250fa1974281a20fcd40d18796))\\n* log grouping for sync output ([b22b381](https://github.com/trivago/github-workflows-skaffold/commit/b22b381efbc3cb9f55ed624f4656252b2f00e756))\\n* log grouping for sync output ([4588608](https://github.com/trivago/github-workflows-skaffold/commit/4588608f1bac060c51504b951638e7e0873465ed))\\n\\n\\n### Bug Fixes\\n\\n* add empty object to release-please manifest ([eac8139](https://github.com/trivago/github-workflows-skaffold/commit/eac81398b2e45c3e2c53b267eb30da664e5c38a6))\\n* add missing backslash for multiline condition ([ea849ef](https://github.com/trivago/github-workflows-skaffold/commit/ea849ef25a680095214986cb4efdd832c6a477ee))\\n* don\'t try to commit when nothing has changed ([5569eeb](https://github.com/trivago/github-workflows-skaffold/commit/5569eeb8ad811af89c8e2a4e96f70f892af1d23c))\\n* invert commit condition ([04bfaec](https://github.com/trivago/github-workflows-skaffold/commit/04bfaec9157a8c459ba9d761dc72c27bd05c0034))\\n* switch all actions to `main` until we have a stable release ([64dd193](https://github.com/trivago/github-workflows-skaffold/commit/64dd19351b4a824daf7a08f2374970b8ec460fc5))\\n* use git fetch to determine if gitops branch exists ([2738bb7](https://github.com/trivago/github-workflows-skaffold/commit/2738bb7a335bacd1c40318a6fec5205a409f6164))\\n\\n\\n### Miscellaneous\\n\\n* **deps:** bump actions/checkout from 3.1.0 to 3.2.0 ([#11](https://github.com/trivago/github-workflows-skaffold/issues/11)) ([9688139](https://github.com/trivago/github-workflows-skaffold/commit/968813984a9ae4535bfc3c632d84f1b53b7816de))\\n* **deps:** bump google-github-actions/release-please-action from 3.6.1 to 3.7.1 ([#10](https://github.com/trivago/github-workflows-skaffold/issues/10)) ([919b2b9](https://github.com/trivago/github-workflows-skaffold/commit/919b2b9ead39e9941da11585e076940e7c705b50))\\n* **deps:** pin actions to v1.0.0 for release ([101ac24](https://github.com/trivago/github-workflows-skaffold/commit/101ac24a8731cae9dcd6f23e913cfb62562da07e))\\n* initial configs ([40bf92a](https://github.com/trivago/github-workflows-skaffold/commit/40bf92a34f358db4e961c904b670ed981653120e))\\n* move editorconfig file to root ([d48954f](https://github.com/trivago/github-workflows-skaffold/commit/d48954fb2fdcb32d1a5b2b86aa814e268e518940))\\n"},{"path":".release-please-manifest.json","mode":"100644","type":"blob","content":"{\\".github/workflows\\":\\"1.0.6\\",\\".github/actions/setup\\":\\"1.0.0\\",\\".github/actions/auth\\":\\"1.0.1\\",\\".github/actions/containerize\\":\\"1.0.1\\",\\".github/actions/render\\":\\"1.0.1\\",\\".github/actions/sync\\":\\"1.0.2\\"}\\n"}],"base_tree":"b506d9067492b5a31706623831fdeb032d823607"}',
    request: { agent: undefined, hook: [Function: bound bound register] }
  }
}
```

### Additional information

I tried setting `bootstrap-sha` to several different commits,
`last-release-sha` and `commit-search-depth` without any noticeable effect.

If this is expected behaviour I's be interested in suggestions how to move components without breaking things.

Also, please let me know if this should rather be posted in https://github.com/googleapis/release-please instead.

Thanks!

**Comment by shinebayar-g** [comment]
Hi, it looks like you're versioning your github actions and workflows. Are you able to do that?

I'm getting this error. I'm curious if it's doable.
```
✔ Got the latest commit tree
Error: release-please failed: Error adding to tree: 57252bf51d55a2962ce8bac1cf8db5e130b80efa
```

**Comment by dsiebel** [comment]
@shinebayar-g That is exactly my problem.
Previously workflows were managed by the root package `"."` and I've been meaning to move that to a dedicated package `.github/workflows`.

I am currently working on a new repository trying to achieve the same:
This time I thought I start directly with a dedicated `.github/workflows` package and I am running into the same error again:

```
✔ Building candidate release pull request for path: .github/workflows
❯ type: simple
❯ targetBranch: main
⚠ No latest release pull request found.
❯ commits: 2
⚠ Setting version for .github/workflows from release-as configuration
✔ Considering: 2 commits
❯ component: workflows
❯ pull request title pattern: undefined
✔ Looking for open release pull requests
✔ found 0 open release pull requests.
✔ Looking for snoozed release pull requests
✔ found 0 snoozed release pull requests.
❯ Fetching .github/workflows/CHANGELOG.md from branch main
❯ Fetching .github/workflows/version.txt from branch main
⚠ file .github/workflows/version.txt did not exist
❯ Fetching .release-please-manifest.json from branch main
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "210eed9af613b4f48ff3362f5a2a62b5a3a743ed".
✔ Skipping branch creation step...
✔ Got the latest commit tree
```

**Comment by dsiebel** [comment]
I managed to work around the issue using the following config:

```json
{
  "release-type": "simple",
  "separate-pull-requests": true,
  "packages": {
    ".": {
      "package-name": "cli",
      "exclude-paths": [
        ".github",
        "tests",
        ".devcontainer.json",
        ".editorconfig",
        ".gitignore",
        ".pre-commit-config.yaml",
        ".release-please-manifest.json",
        "justfile",
        "mkdocs.yaml",
        "README.md",
        "release-please-config.json",
        "renovate.json"
      ]
    },
    ".github/actions/apply": {
      "package-name": "action-apply"
    },
    ".github/actions/setup": {
      "package-name": "action-setup"
    },
	"A BUNCH MORE ACTIONS HERE": { ... },
    ".github": {
      "package-name": "workflows",
      "exclude-paths": [
        "actions",
        "CODEOWNERS",
        "workflows/ci.yaml",
        "workflows/release-please.yaml"
      ]
    }
  }
}

```

**Comment by shinebayar-g** [comment]
Yeah I am doing **exactly** the same thing for now and using `exclude-paths` for `.github/actions/*` and `.` packages.

```json
...
  "packages": {
    ".github": {
      "package-name": "workflows",
      "release-type": "simple"
    },
...
```

This is a little bit ugly, but at least works for now ... Thanks for sharing your full configuration.

**Comment by shinebayar-g** [comment]
I haven't tested `.github/actions` yet. This is all I have in my packages

```json
  "packages": {
    ".github": {
      "package-name": "workflows",
      "release-type": "simple",
      "exclude-paths": ["workflows/release.yaml"] # can't remember if it was needed
    },
    "packages/my-package": {
      "package-name": "my-package",
      "release-type": "node"
    }
  },
```

Are you running into issues with `.github/actions`?

**Comment by chingor13** [comment]
Touching files in `.github/workflows` requires additional permissions. https://github.com/google-github-actions/release-please-action#workflow-permissions

You may need to grant additional permissions to your access token being used. The built-in token may not work and you might need to create a personal access token.
